remix-tests/src/compiler.ts

```diff
@@ -21,39 +21,84 @@ function writeTestAccountsContract (accounts: string[]) {
     return testAccountContract.replace('>accounts<', body)
 }
 
+/**
+ * @dev Check if path includes name of a remix test file
+ * @param path file path to check
+ */
+
+function isRemixTestFile(path: string) {
+    return ['tests.sol', 'remix_tests.sol', 'remix_accounts.sol'].some(name => path.includes(name))
+}
+
+/**
+ * @dev Process file to prepare sources object to be passed in solc compiler input
+ * 
+ * See: https://solidity.readthedocs.io/en/latest/using-the-compiler.html#input-description
+ * 
+ * @param filePath path of file to process
+ * @param sources existing 'sources' object in which keys are the "global" names of the source files and 
+ *                value is object containing content of corresponding file with under key 'content'
+ * @param isRoot True, If file is a root test contract file which is getting processed, not an imported file
+ */
+
+function processFile(filePath: string, sources: any, isRoot: boolean = false) {
+    const importRegEx: RegExp = /import ['"](.+?)['"];/g;
+    let group: RegExpExecArray| null = null;
+    const isFileAlreadyInSources: boolean = Object.keys(sources).includes(filePath)
+
+    // Return if file is a remix test file or already processed
+    if(isRemixTestFile(filePath) || isFileAlreadyInSources)
+        return
+
+    let content: string = fs.readFileSync(filePath, { encoding: 'utf-8' });
+    const testFileImportRegEx: RegExp = /^(import)\s['"](remix_tests.sol|tests.sol)['"];/gm
+
+    // import 'remix_tests.sol', if file is a root test contract file and doesn't already have it 
+    if (isRoot && filePath.includes('_test.sol') && regexIndexOf(content, testFileImportRegEx) < 0) {
+        const includeTestLibs: string = '\nimport \'remix_tests.sol\';\n'
+        content = includeTestLibs.concat(content)
+    }
+    sources[filePath] = {content};
+    importRegEx.exec(''); // Resetting state of RegEx
+
+    // Process each 'import' in file content
+    while (group = importRegEx.exec(content)) {
+        const importedFile: string = group[1];
+        const importedFilePath: string = path.join(path.dirname(filePath), importedFile);
+        processFile(importedFilePath, sources)
+    }
+}
+
 const userAgent = (typeof (navigator) !== 'undefined') && navigator.userAgent ? navigator.userAgent.toLowerCase() : '-'
 const isBrowser = !(typeof (window) === 'undefined' || userAgent.indexOf(' electron/') > -1)
 
 // TODO: replace this with remix's own compiler code
 export function compileFileOrFiles(filename: string, isDirectory: boolean, opts: any, cb: Function) {
     let compiler: any
-    let accounts = opts.accounts || []
-    const sources = {
+    let accounts: any[] = opts.accounts || []
```

Comment:
You're not reassigning it. You can use `const`. Also, if accounts are just addresses, use `string[]`

Reply:
done